Invalid use of sizeof() on pointers in Ed448 tests

When building with clang and `-Werror` (and the default of `--enable-crypttests`) I get the following errors:

```
wolfcrypt/test/test.c:23114:42: error: 'XMEMCMP' call operates on objects of type 'const byte' (aka 'const unsigned char') while the size is based on a different type 'const byte *' (aka 'const unsigned char *') [-Werror,-Wsizeof-pointer-memaccess]
        if (XMEMCMP(out, sigs[i], sizeof(sigs[i])))
                         ~~~~~~~         ^~~~~~~
wolfcrypt/test/test.c:23114:42: note: did you mean to provide an explicit length?
        if (XMEMCMP(out, sigs[i], sizeof(sigs[i])))
                                         ^~~~~~~
wolfcrypt/test/test.c:23146:38: error: 'XMEMCMP' call operates on objects of type 'const byte' (aka 'const unsigned char') while the size is based on a different type 'const byte *' (aka 'const unsigned char *') [-Werror,-Wsizeof-pointer-memaccess]
    if (XMEMCMP(out, sigs[0], sizeof(sigs[0])))
                     ~~~~~~~         ^~~~~~~
wolfcrypt/test/test.c:23146:38: note: did you mean to provide an explicit length?
    if (XMEMCMP(out, sigs[0], sizeof(sigs[0])))
                                     ^~~~~~~
wolfcrypt/test/test.c:23167:38: error: 'XMEMCMP' call operates on objects of type 'const byte' (aka 'const unsigned char') while the size is based on a different type 'const byte *' (aka 'const unsigned char *') [-Werror,-Wsizeof-pointer-memaccess]
    if (XMEMCMP(out, sigs[0], sizeof(sigs[0])))
                     ~~~~~~~         ^~~~~~~
wolfcrypt/test/test.c:23167:38: note: did you mean to provide an explicit length?
    if (XMEMCMP(out, sigs[0], sizeof(sigs[0])))
                                     ^~~~~~~
3 errors generated.
```

These are from the newly added Ed448 test cases.  Using a pointer to an array in `sizeof()` and not the array itself  causes these checks to only compare the first 8 bytes (on x64) of the signatures.

--- a/wolfcrypt/test/test.c
+++ b/wolfcrypt/test/test.c
@@ -5747,7 +5747,8 @@ int des3_test(void)
 
 #ifndef NO_AES
 
-#if defined(WOLFSSL_AES_OFB) || defined(WOLFSSL_AES_CFB)
+#if defined(WOLFSSL_AES_OFB) || defined(WOLFSSL_AES_CFB) || \
+    defined(WOLFSSL_AES_XTS)
 #if defined(OPENSSL_EXTRA) && !defined(HAVE_SELFTEST) && !defined(HAVE_FIPS)
 /* pass in the function, key, iv, plain text and expected and this function
  * tests that the encryption and decryption is successful */
@@ -22833,6 +22834,7 @@ int ed448_test(void)
     };
 
     static const byte* sigs[] = {sig1, sig2, sig3, sig4, sig5, sig6};
+    #define SIGSZ sizeof(sig1)
 
     static const byte msg1[]  = { };
     static const byte msg2[]  = { 0x03 };
@@ -23111,7 +23113,7 @@ int ed448_test(void)
                                 NULL, 0) != 0 || verify != 1)
             return -11401 - i;
 
-        if (XMEMCMP(out, sigs[i], sizeof(sigs[i])))
+        if (XMEMCMP(out, sigs[i], SIGSZ))
             return -11411 - i;
 #endif /* HAVE_ED448_VERIFY */
     }
@@ -23143,7 +23145,7 @@ int ed448_test(void)
     if (wc_ed448_sign_msg(msgs[0], msgSz[0], out, &outlen, &key3, NULL, 0) != 0)
         return -11451 - i;
 
-    if (XMEMCMP(out, sigs[0], sizeof(sigs[0])))
+    if (XMEMCMP(out, sigs[0], SIGSZ))
         return -11461 - i;
 
 #if defined(HAVE_ED448_VERIFY)
@@ -23164,7 +23166,7 @@ int ed448_test(void)
     if (wc_ed448_sign_msg(msgs[0], msgSz[0], out, &outlen, &key3, NULL, 0) != 0)
         return -11491 - i;
 
-    if (XMEMCMP(out, sigs[0], sizeof(sigs[0])))
+    if (XMEMCMP(out, sigs[0], SIGSZ))
         return -11501 - i;
 
     wc_ed448_free(&key3);

